src/Stratis.Bitcoin/Consensus/Rules/ConsensusRule.cs

```diff
@@ -24,31 +23,41 @@ public virtual void Initialize()
     }
 
     /// <summary>An abstract rule for implementing consensus rules.</summary>
-    public abstract class SyncConsensusRule : ConsensusRuleBase, ISyncBaseConsensusRule
+    public abstract class SyncConsensusRule : ConsensusRuleBase
```

Comment:
And now, why not interface?

Reply:
You mean why not inherit from IConsensusRuleBase? 
Because abstract class already has 3 members already which all rules need.